lib/AST/Evaluator.cpp

```diff
@@ -75,35 +73,40 @@ void Evaluator::emitRequestEvaluatorGraphViz(llvm::StringRef graphVizPath) {
   printDependenciesGraphviz(out);
 }
 
-bool Evaluator::checkDependency(const AnyRequest &request) {
+bool Evaluator::checkDependency(const ActiveRequest &request) {
   if (buildDependencyGraph) {
+    // Insert the request into the dependency graph if we haven't already.
+    auto req = AnyRequest(request);
+    dependencies.insert({req, {}});
+
     // If there is an active request, record it's dependency on this request.
     if (!activeRequests.empty())
-      dependencies[activeRequests.back()].push_back(request);
+      dependencies[AnyRequest(activeRequests.back())].push_back(req);
```

Comment:
It's no big deal for now, but a little micro-optimization here and elsewhere could be using `find_as` to do lookups with ActiveRequests instead of AnyRequests.

Reply:
@CodaFi Good idea! How does this look?

Reply:
Fantastic.  Let’s get another CI run and Doug’s rubber stamp.